test/unit/test_util.c

```diff
@@ -62,7 +63,76 @@ bool test_leading_zeros(void) {
 	mu_end;
 }
 
+bool test_set_u(void) {
+	SetU *set_u = set_u_new();
+	set_u_add(set_u, 0x5050505);
+	set_u_add(set_u, 0x5050505);
+	set_u_add(set_u, 0x6060606);
+	set_u_add(set_u, 0x7070707);
+	set_u_add(set_u, 0x7070707);
+	mu_assert_eq(set_u_size(set_u), 3, "Length wrong.");
+	mu_assert_true(set_u_contains(set_u, 0x5050505), "Value was not added.");
+	mu_assert_true(set_u_contains(set_u, 0x6060606), "Value was not added.");
+	mu_assert_true(set_u_contains(set_u, 0x7070707), "Value was not added.");
+
+	set_u_delete(set_u, 0x7070707);
+	mu_assert_false(set_u_contains(set_u, 0x7070707), "Value was not deleted.");
+	mu_assert_eq(set_u_size(set_u), 2, "Length wrong.");
+
+	// Double delete
+	set_u_delete(set_u, 0x7070707);
+	mu_assert_eq(set_u_size(set_u), 2, "Length wrong.");
+
+	size_t x = 0;
+	SetUIter it;
+	set_u_foreach (set_u, it) {
+		x++;
+		bool matches = it.v == 0x5050505 || it.v == 0x6060606;
+		mu_assert_true(matches, "Set contained ill-formed value.");
+	}
```

Comment:
I would prefer `rz_list_foreach` like variant with separate `value` variable:
```suggestion
	SetUIter it;
	ut64 val;
	set_u_foreach (set_u, it, val) {
		x++;
		bool matches = val == 0x5050505 || val == 0x6060606;
		mu_assert_true(matches, "Set contained ill-formed value.");
	}
```

Reply:
@ret2libc your opinion?

Reply:
Don't really have strong feelings about it. Separated variable means the type is more explicit, which is probably worth the trade against the extra line. But as long as we have a `foreach` without a callback, I am happy.

Reply:
https://github.com/rizinorg/rizin/blob/ea460e54c28d7e44d123fe59bf123fda12e3bcd2/librz/include/rz_list.h#L35

Adapted for your case: `(val = iter.v, 1)`

Reply:
I think the one with the value is better.

Reply:
It will be more important for the SetP, where we have a room for type ambiguity.

Reply:
i think the value should always be a pointer. so you can read and write the value.

Reply:
> i think the value should always be a pointer. so you can read and write the value.

For HT, yes. But in case of sets value is a key.